### PR TITLE
feat: #5 support server side storage upload

### PR DESCRIPTION
**Pull request by thebengeu**

### CODE REVIEW

**Comment by inian** [comment]
Can you fix the tests Beng? Rest looks good

**Comment by thebengeu** [comment]
Made substantial changes to accommodate the update operation, please take a look.

File: src/lib/StorageFileApi.ts [hunk 1]
```diff
@@ -28,34 +30,56 @@ export class StorageFileApi {
...
+  private async uploadOrUpdate(
+    method: 'POST' | 'PUT',
     path: string,
-    file: File,
+    fileBody:
+      | ArrayBuffer
+      | ArrayBufferView
+      | Blob
+      | File
+      | FormData
```
Comment (inian) [inline src/lib/StorageFileApi.ts:51]: Nice! Users can pass in FormData in directly now

**Review by inian — APPROVED** [review]

**Pull request merged**